### ISSUE TITLE
Add source_transaction to the transfer object

### ISSUE THREAD

**Issue by Magrangs** [stripe/stripe-dotnet#960]
The transfer object contains a source_transaction field (https://stripe.com/docs/api#transfer_object) .

Which is the "ID of the charge or payment that was used to fund the transfer. If null, the transfer was funded from the available balance"

This is needed to connect the transfer back to the charge. **It looks like this has been added to the StripeTransferCreateOptions class but it is not present on the StripeTransfer class**

Thanks

**Comment by remi-stripe** [comment]
@Magrangs Thanks for raising, I had forgotten about this one. I just pushed a fix in https://github.com/stripe/stripe-dotnet/pull/962 so hopefully we can make a new release in the next few days.

**Comment by Magrangs** [comment]
Ah great, thanks for that @remi-stripe . Is there a way to get the charge objects and the transfer objecs in one hit (from the list endpoint)? I need information from both but what I don't want to do is call the api again for each transfer object that I have just to get the charge object associated with it

**Comment by remi-stripe** [comment]
@Magrangs correct, the library has a limited support for Expand right now and you would need to call the API for each one after that (or cache most of this on your end).

Supporting this field would be a bit complex as `source_transaction` can be various types. You can see how many in the Go library where it's been implemented [here](https://github.com/stripe/stripe-go/blob/6e459d8e0811d17fa76afc4286d544ad35b731b5/balance.go#L148).

It's definitely not impossible but I don't think we should ship this so quickly. We should first do a full pass on expandable fields and ship expand for all at once to avoid making confusing incremental updates.

**Comment by Magrangs** [comment]
@remi-stripe Ah ok no worries, I will write a service on my end then in the mean time. I assume you don't have an idea on time-frames yet?

Many thanks.

**Comment by remi-stripe** [comment]
@Magrangs I don't have a clear timeline for supporting full Expand. We would definitely appreciate a PR if you are planning to work on supporting export for `SourceTransaction` though!

**Comment by Magrangs** [comment]
@remi-stripe Just forked the repo and stepping through some of the tests. One thing I noticed is that the tests are actually creating a real http client and calling the stripe endpoints. They appear to be integration tests rather than unit tests? I may be missing something however? Getting an issue with the api key as it's null, just working out where to stick it to get the tests to run. Cheers.

**Comment by remi-stripe** [comment]
@Magrangs Correct, this library hits the API for all tests, they are not real unit tests. Some of our libs (like Node.js or Ruby) support real unit tests but others hit the API. It is something we're hoping to fix in the next few months though but no promises.

You need to set your API key when you run the tests but it depends a bit how you're running them. I am not really familiar with .NET development myself and have been learning recently and using this on OSX. When I run a specific file for example, here's what I do on the command line

```
STRIPE_TEST_SK=sk_test_XXXX dotnet test --filter FullyQualifiedName~Stripe.Tests.Xunit.creating_and_updating_transfers ./src/Stripe.Tests.XUnit/Stripe.Tests.XUnit.csproj /p:TargetFramework=netcoreapp1.1
```

Hope this helps go in the right direction. We're hoping to improve the readme to cover most of this but haven't had time to look into this one just yet so it will require a bit of trial and error on your end or some patience until we can implement this fully.

**Comment by jaymedavis** [comment]
I would recommend creating a json converter to handle this on StripeTransfer. [This example](https://github.com/stripe/stripe-dotnet/blob/master/src/Stripe.net/Infrastructure/JsonConverters/SourceConverter.cs) should be helpful. 

1) create a set of new properties on StripeTransfer: SourcePayment, SourceTransfer, SourceX, where X = anything that can be a source
2) create a new json converter that interrogates the return json's property field, and uses the `Mapper` to set the field, as it does in the SourceConverter

Hope this helps!

**Comment by ob-stripe** [comment]
This was fixed by @remi-stripe a while ago, in #1139, and released as 15.0.0.
